### PR TITLE
Add consumer metrics

### PR DESCRIPTION
**Pull request by Shreshth3**
-updated npm package for backend (added trackConsumer function)
-updated server/server.js to handle consumer data
-restructured frontend to be more DRY
-added frontend components to display consumer metrics
-reconfigured Web Socket functionality to handle consumer data

### CODE REVIEW

**Review by ymturan — APPROVED** [review]

File: app/components/App.jsx [hunk 1]
```diff
@@ -1,23 +1,22 @@
 import React, { Component } from 'react';
 import Dashboard from './Dashboard';
 import Homepage from './Homepage';
+// import Icon from './assets/kafkaq_logoTest.svg';
```
Comment (Jbaren01) [inline app/components/App.jsx]: we can delete these

File: app/components/App.jsx [hunk 2]
```diff
@@ -1,23 +1,22 @@
...
-//   key: '9cf4213d3cc82809c8344f68d1ea107f',
-//   base: 'https://api.openweathermap.org/data/2.5/'
-// };
-
 class App extends Component {
   render() {
     return (
       <div className="app">
         <div className="title">
+          {/* <img height={25} src={logo} alt="logo" /> */}
```
Comment (Jbaren01) [inline app/components/App.jsx]: delete this

File: app/components/App.jsx [hunk 3]
```diff
@@ -1,23 +1,22 @@
...
           <h2>kafkaQ</h2>
           <h1>Dashboard</h1>
         </div>
         <div className="dashboard-container">
           <Dashboard />
+          {/* <Icon width={50} height={50} /> */}
```
Comment (Jbaren01) [inline app/components/App.jsx]: delete this

File: app/components/Dashboard.jsx [hunk 4]
```diff
@@ -38,72 +64,114 @@ class Dashboard extends Component {
...
+        },
+      };
+
+      this.setState({
+        ...this.state,
+        consumer: consumerData,
+      });
+    });
 
     // socket.emit('message', {
```
Comment (Jbaren01) [inline app/components/Dashboard.jsx]: can we clean this up/delete?

File: app/components/Icon.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
```
Comment (Jbaren01) [inline app/components/Icon.jsx]: i thought this had been deleted? we should remove

File: app/components/SystemData.jsx [hunk 6]
```diff
@@ -1,94 +1,87 @@
 import React, { Component } from 'react';
-import StopLight from './metrics/StopLight';
-import Graph from './metrics/Graph';
+import StopLight from './old-metrics/StopLight';
+import Graph from './old-metrics/Graph';
 import { time } from 'console';
```
Comment (Jbaren01) [inline app/components/SystemData.jsx]: did we ever figure out what this does or is?

**Review by Jbaren01 — APPROVED** [review]

**Pull request merged**